ofrak_core/ofrak/service/data_service.py

```diff
@@ -1,1464 +1,402 @@
-                )
-            if after_data_id is not None:
-                if index == 0:
-                    raise AmbiguousOrderError(
-                        f"The {child_range} was hinted to be after the child "
-                        f"{after_data_id.hex()} but there is no child to the left"
-                    )
-                _, left_child_node = self._children[index - 1]
-                if left_child_node.model.id != after_data_id:
-                    hinted_after_index = self._children_index_by_id[after_data_id]
-                    _, hinted_after_node = self._children[hinted_after_index]
-                    raise AmbiguousOrderError(
-                        f"The {child_range} is after {left_child_node} but it was "
-                        f"hinted to be after the child {hinted_after_node}"
-                    )
-                return index - 1, DataRangePosition.AFTER
-            if before_data_id is not None:
-                if index >= len(self._children):
-                    raise AmbiguousOrderError(
-                        f"The {child_range} was hinted to be after the child "
-                        f"{before_data_id.hex()} but there is no child to the right"
-                    )
-                _, right_child_node = self._children[index]
-                if right_child_node.model.id != before_data_id:
-                    raise AmbiguousOrderError(
-                        f"The {child_range} is before {right_child_node} but it was "
-                        f"hinted to be before the child {before_data_id.hex()}"
-                    )
-                return index, DataRangePosition.BEFORE
-            return index, relative_position
-        else:
-            _, child_node = self._children[index]
-            raise ValueError(
-                f"Received unexpected relative_position value {relative_position}"
-                f" for range {child_range}, child {child_node}, and node ID"
-                f" {self.model.id.hex()}"
-            )
+        return new_model
 
-    def get_range_index(
+    async def create_mapped(
         self,
-        child_range: Range,
-        within_data_id: Optional[bytes] = None,
-        after_data_id: Optional[bytes] = None,
-        before_data_id: Optional[bytes] = None,
-    ) -> Tuple[int, DataRangePosition]:
-        """
-        Return the index of an existing node and the enum value describing the
-        relation between ``child_range`` and that node.
-        """
-        if child_range.start < 0 or child_range.end > self.model.range.length():
+        data_id: DataId,
+        parent_id: DataId,
+        mapped_range: Range,
+    ) -> DataModel:
+        if data_id in self._model_store:
+            raise AlreadyExistError(f"A model with {data_id.hex()} already exists!")
+
+        parent_model = self._get_by_id(parent_id)
+        mapped_range = mapped_range.translate(parent_model.range.start)
+        if mapped_range.end > parent_model.range.end:
```

Comment:
I find all of these mapped ranges confusing. It took me a while to figure out why the translation here is necessary.

The "mapped_range" passed to this method is the range of the data in the parent's data.

This line:
```
mapped_range = mapped_range.translate(parent_model.range.start)
```

Seems to then get the range of the data mapped into the root data (which may or may not be the parent.

What do you think of
- Renaming the parameter for `create_mapped` to `range_in_parent`
- Changing the following:
```
mapped_range = mapped_range.translate(parent_model.range.start)
```
to
```
range_in_root = range_in_parent.translate(parent_model.range.start)
```
?